core/drivers/atmel_shdwc.c

```diff
@@ -178,8 +196,27 @@ static TEE_Result atmel_shdwc_probe(const void *fdt, int node,
 	return sama5d2_pm_init(fdt, shdwc_base);
 }
 
+static const struct shdwc_compat sama5d2_compat = {
+	.shdwc_as = 0,
+	.ddrc = {
+		.type_offset = AT91_DDRSDRC_MDR,
+		.type_mask = AT91_DDRSDRC_MD,
```

Comment:
Only `type_offset` is used, it is a bit weird, the code looks incomplete. I suppose these values are intended to be used in the `io_read32()` call in the probe function? Then please update accordingly.